commonItems/UnitTests/ColorTests.cs

```diff
@@ -0,0 +1,622 @@
+            var color = new Foo(reader).color;
+
+            Assert.Equal(64, color.R);
+            Assert.Equal(128, color.G);
+            Assert.Equal(128, color.B);
+
+            Assert.Equal(0.5, color.H, decimalPlaces);
+            Assert.Equal(0.5, color.S, decimalPlaces);
+            Assert.Equal(0.5, color.V, decimalPlaces);
+
+            Assert.Equal(" more text", reader.ReadLine());
+        }
+
+        [Fact]
+        public void ColorCanBeInitializedInRgbFromLongerStream() {
+            var reader = new BufferedReader("= { color = rgb { 64 128 128 } } more text");
+            var color = new Foo(reader).color;
+
+            Assert.Equal(64, color.R);
+            Assert.Equal(128, color.G);
+            Assert.Equal(128, color.B);
+
+            Assert.Equal(0.5, color.H, decimalPlaces);
+            Assert.Equal(0.5, color.S, decimalPlaces);
+            Assert.Equal(0.5, color.V, decimalPlaces);
+
+            Assert.Equal(" more text", reader.ReadLine());
+        }
+
+        [Fact]
+        public void ColorCanBeInitializedInHsvFromLongerStream() {
+            var reader = new BufferedReader("= { color = hsv { 0.5 0.5 0.5 } } more text");
+            var color = new Foo(reader).color;
+
+            Assert.Equal(63, color.R);
+            Assert.Equal(127, color.G);
+            Assert.Equal(127, color.B);
+
+            Assert.Equal(0.5, color.H, decimalPlaces);
+            Assert.Equal(0.5, color.S, decimalPlaces);
+            Assert.Equal(0.5, color.V, decimalPlaces);
+
+            Assert.Equal(" more text", reader.ReadLine());
+        }
+
+        [Fact] public void ColorCanBeOutputInUnspecifiedColorSpace() {
+            var color = new Color(new int[] { 64, 128, 128 });
+            Assert.Equal("= { 64 128 128 }", color.Output());
+        }
+
+        [Fact]
+        public void ColorCanBeOutputInRgbColorSpace() {
+            var color = new Color(new int[] { 64, 128, 128 });
+            Assert.Equal("= rgb { 64 128 128 }", color.OutputRgb());
+        }
+
+        [Fact]
+        public void ColorCanBeOutputInHexColorSpace() {
+            var color = new Color(new int[] { 64, 128, 128 });
+            Assert.Equal("= hex { 408080 }", color.OutputHex());
+        }
+
+        [Fact]
+        public void ColorCanBeOutputInHexColorSpacePreservingZeroes() {
+            var color = new Color(new int[] { 0, 0, 0 });
+            Assert.Equal("= hex { 000000 }", color.OutputHex());
+        }
+
+        [Fact]
+        public void ColorCanBeOutputInHsvColorSpace() {
+            var color = new Color(new int[] { 64, 128, 128 });
+            Assert.Equal("= hsv { 0.5 0.5 0.5 }", color.OutputHsv());
+        }
+
+        [Fact]
+        public void ColorCanBeOutputInHsv360ColorSpace() {
+            var color = new Color(new int[] { 64, 128, 128 });
+            Assert.Equal("= hsv360 { 180 50 50.2 }", color.OutputHsv360());
```

Comment:
I've searched through CK3 files, CK3 wiki and https://rakaly.com/blog/a-tour-of-pds-clausewitz-syntax. It seems the hsv360 values should be integers.

Reply:
I guess I will just use Math.Round to round to nearest integer. I still don't know why the C++ version outputs 50 instead of 50.2 without having to round.